contracts/drafts/ERC777/ERC777.sol

```diff
@@ -247,30 +312,33 @@ contract ERC777 is IERC777 {
      * @param amount uint256 amount of tokens to transfer
      * @param userData bytes extra information provided by the token holder (if any)
      * @param operatorData bytes extra information provided by the operator (if any)
+     * @param avoidLockingTokens if true, contract recipients are required to implement ERC777TokensRecipient
      */
     function _send(
         address operator,
         address from,
         address to,
         uint256 amount,
         bytes memory userData,
-        bytes memory operatorData
+        bytes memory operatorData,
+        bool avoidLockingTokens
```

Comment:
Not a fan of this being a boolean because at the callsite it looks very cryptic. Also not a fan of the argument name.

What do you think about using an `enum` for maximum explicitness?
```
enum ERC777TokensRecipientCheck {
    Required,
    NotRequired
}
```

What's not so good about this is that the enum type is made available to derived contracts, but we can add a `__unstable__` prefix or something like that. Not sure if all this is worth it...